docs-source/content/userguide/managing-domains/domain-lifecycle/restarting.md

```diff
@@ -3,116 +3,123 @@ title: "Restarting"
-      * You must either supply a new image name in the domain resource or change the domain resource `restartVersion` in order to force the operator to reload the configuration.
-      * With either of these two changes, the operator will rerun the domain's introspector job, which will verify and apply the new configuration. If the introspector job's configuration verification succeeds, then it will subsequently roll (restart) the pods; if the job fails, then a roll will not occur.
-      * If you change other fields that typically cause a restart, such as `volumes`, `env`, and such, then the introspector job will not rerun and a rolling restart will proceed without loading the configuration changes.
+    * If you are supplying updated models or Secrets for a running domain, and you want the configuration updates to take effect using a rolling restart, then do one of the following:
+      * Supply a new value for the `image` field in the Domain or any of the other [fields affecting WebLogic Server instance Pod generation]({{< relref "/userguide/managing-domains/domain-lifecycle/startup/_index.md#fields-that-cause-servers-to-be-restarted" >}}).
+      * Change the Domain `restartVersion` field. This will cause the operator to restart all running servers and, prior to the restarts, the operator will introspect any new configuration.
+      * Change the Domain `introspectVersion` field. This will cause the operator to introspect any new configuration and, if needed, restart servers to use that new configuration.
 
 * **Domain in PV:**
-For a domain home on PV, the type of restart needed to apply the changes depends on the nature of the WebLogic configuration change:
-    * Changes to parts of the WebLogic configuration that the operator introspects, require a full restart, even if the changes are dynamic.
-      The following are the types of changes to the WebLogic Server configuration that the operator introspects:
-        * Adding or removing a cluster, server, dynamic server, or network access point
+
+For Domain in PV, the type of restart needed depends on the nature of the WebLogic domain configuration change:
+    * With operator version 3.0.0, domain configuration changes that add clusters, either configured or dynamic, cluster member servers, or standalone servers can now be performed dynamically. This support requires that the new clusters or servers are added to the domain configuration and then [triggering the operator's introspection]({{< relref "/userguide/managing-domains/domain-lifecycle/introspection.md" >}}) of that new configuration.
+    * Other changes to parts of the domain configuration that the operator introspects, require a full restart, even if the changes are dynamic.
+      The following are the other types of changes to the domain configuration that the operator introspects:
+        * Adding or removing a network access point
```

Comment:
I think we still don't support Adding/removing a server to/from an existing cluster.

Reply:
The owning bullet's two sentences are confusing when read together (Other changes... and The following ...).  They conflate what requires a full restart with what the operator introspects (they're not the same thing).

Reply:
How do you suggest that I update the wording? I don't really understand how we could not support adding a server to an existing cluster. Is your concern about whether already running servers would be able to properly address the new server or something else?

Reply:
Adding a server to an existing cluster has the following concerns:
- This hasn't been tested.  It's not known how well this works in combination with a dynamic (or rolling?) update.
- The admin server and existing servers need to know the _overridden_ address/port of the new server, for JNDI, JTA, RMI stub load balancing, etc, purposes.  
- It's not clear if these fields are fully dynamic for mbean/configuration purposes, or for their individual purposes.

Reply:
Do we document any of these limitations or problems for on-premise WebLogic? Just to clarify, I'm not doubting that there are problems here. It's just very ugly to only say that you can't add a server to an existing cluster for all use cases because there exist use cases where this may be problematic.

Reply:
We don't know which use cases work well (if any) - there's been no testing IIRC.

The WL documentation for situational configuration doesn't seem to mention situations where network configuration of newly added sister-servers in a cluster is dynamically overridden: https://docs.oracle.com/middleware/12213/wls/DOMCF/changes.htm#DOMCF172

Reply:
Okay, I'll try and come up with something.

Reply:
The 'something' LGTM.